packages/api/src/ontology/ActionDefinition.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface ActionDefinition<
+  A extends String,
+  K extends String,
+> {
+  apiName: A;
+  description?: string;
+  displayName?: string;
+  parameters: Record<string, ActionParameterDefinition<K>>;
```

Comment:
Can we have different K's on different parameters?

Reply:
Yup!

Reply:
Oh right but K can be several strings not just one. I always forget that.